packages/cli/lib/DevServerManager.js

```diff
@@ -1,28 +1,35 @@
 const express = require('express');
 const bodyParser = require('body-parser');
 const cors = require('cors');
-const { walk } = require('@hubspot/cli-lib/lib/walk');
+const httpClient = require('@hubspot/cli-lib/http');
+const { logger } = require('@hubspot/cli-lib/logger');
 const { getProjectDetailUrl } = require('./projects');
+const { COMPONENT_TYPES } = require('./projectStructure');
 const { i18n } = require('./lang');
 const { EXIT_CODES } = require('./enums/exitCodes');
-const { logger } = require('@hubspot/cli-lib/logger');
+const { promptUser } = require('./prompts/promptUtils');
 
 const i18nKey = 'cli.lib.DevServerManager';
 
 const DEFAULT_PORT = 8080;
-
 class DevServerManager {
   constructor() {
     this.initialized = false;
+    this.started = false;
+    this.componentsByType = {};
     this.server = null;
     this.path = null;
     this.devServers = {};
+    this.debug = false;
   }
 
   safeLoadServer() {
     try {
       const { DevModeInterface } = require('@hubspot/ui-extensions-dev-server');
-      this.devServers['uie'] = DevModeInterface;
+      this.devServers['uie'] = {
```

Comment:
I'm grouping servers with component types. This makes it easier to know whether or not we should even bother calling the lifecycle methods for certain servers.

Reply:
I know this isn't new code but seems like it could be a good candidate for a constant / enum. `SERVER_KEYS.UIE` or something

Reply:
👍 done in [0df9f69](https://github.com/HubSpot/hubspot-cli/pull/892/commits/0df9f69e75b68e378f92bdb5e4bc291434a94743)